LocalNotificationHelper/S2MNotificationHelper.m

```diff
@@ -0,0 +1,254 @@
+{
+    static NSString *documentsPath;
+    
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSArray *documentsPaths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+        documentsPath = [documentsPaths objectAtIndex:0];
+        
+        documentsPath = [documentsPath stringByAppendingPathComponent:CACHE_FOLDER_NAME];
+        if (![[NSFileManager defaultManager] fileExistsAtPath:documentsPath]) {
+            [[NSFileManager defaultManager] createDirectoryAtPath:documentsPath withIntermediateDirectories:NO attributes:nil error:nil];
+        }
+        [self addSkipBackupAttributeToItemAtFilePath:documentsPath];
+    });
+    
+    return documentsPath;
+}
+
++ (NSString *)cachePathWithKey:(NSString *)key
+{
+    NSString *cacheHash = [key s2m_hashString];
+    NSString *archivePath = [NSString stringWithFormat:@"%@/%@", [self cacheFolderPath], cacheHash];
+    return archivePath;
+}
+
++ (BOOL)hasCacheForKey:(NSString *)key
+{
+    NSString *archivePath = [self cachePathWithKey:key];
+    if ([[NSFileManager defaultManager] fileExistsAtPath:archivePath]) {
+        return YES;
+    }
+    return NO;
+}
+
++ (void)cacheObject:(id<NSCoding>)obj forKey:(NSString *)key
+{
+    NSString *archivePath = [self cachePathWithKey:key];
+    [NSKeyedArchiver archiveRootObject:obj toFile:archivePath];
+}
+
++ (id<NSCoding>)cacheForKey:(NSString *)key
+{
+    NSString *archivePath = [self cachePathWithKey:key];
+    
+    return !archivePath ? nil : [NSKeyedUnarchiver unarchiveObjectWithFile:archivePath];
+}
+
++ (void) removeAllCaches
+{
+    NSFileManager* fileManager = [[NSFileManager alloc] init];
+    NSString *folderPath = [self cacheFolderPath];
+    NSDirectoryEnumerator* enumerator = [fileManager enumeratorAtPath:folderPath];
+    NSError* err = nil;
+    BOOL res;
+    
+    NSString* file;
+    while (file = [enumerator nextObject]) {
+        res = [fileManager removeItemAtPath:[folderPath stringByAppendingPathComponent:file] error:&err];
+        if (!res && err) {
+            NSLog(@"Can't remove Notification: %@", err);
+        }
+    }
+}
+
++ (NSArray *)allCaches
+{
+    NSMutableArray *caches = [NSMutableArray array];
+    NSFileManager* fileManager = [[NSFileManager alloc] init];
+    NSString *folderPath = [self cacheFolderPath];
+    NSDirectoryEnumerator* enumerator = [fileManager enumeratorAtPath:folderPath];
+    NSError* err = nil;
+    
+    NSString* file;
+    while (file = [enumerator nextObject]) {
+        id cahceObject = [NSKeyedUnarchiver unarchiveObjectWithFile:[folderPath stringByAppendingPathComponent:file]];
+        if (!cahceObject) {
+            NSLog(@"Can't read cache object: %@", err);
+        } else {
+            [caches addObject:cahceObject];
+        }
+    }
+    return caches;
+}
+
+#pragma mark - public API
+
++ (UILocalNotification *)localNotificationForKey:(NSString *)key withRemoteNotification:(NSDictionary *)userInfo
```

Comment:
rename method to match parameters,
` +(UILocalNotification*)localNotificationForKey:userInfo:`